deps/readies/paella/setup.py

```diff
@@ -0,0 +1,156 @@
+
+import os
+import sys
+import tempfile
+from .platform import OnPlatform, Platform
+
+#----------------------------------------------------------------------------------------------
+
+class Runner:
+    def __init__(self, nop=False):
+        self.nop = nop
+
+    def run(self, cmd, output_on_error=False):
+        print(cmd)
+        sys.stdout.flush()
+        if self.nop:
+            return
+        if output_on_error:
+            fd, temppath = tempfile.mkstemp()
+            os.close(fd)
+            cmd = "{{ {}; }} >{} 2>&1".format(cmd, temppath)
+        rc = os.system(cmd)
+        if rc > 0:
+            if output_on_error:
+                os.system("cat {}".format(temppath))
+                os.remove(temppath)
+            eprint("command failed: " + cmd)
+            sys.stderr.flush()
+            sys.exit(1)
+
+    def has_command(self, cmd):
+        return os.system("command -v " + cmd + " > /dev/null") == 0
+
+#----------------------------------------------------------------------------------------------
+
+class RepoRefresh(OnPlatform):
+    def __init__(self, runner):
+        OnPlatform.__init__(self)
+        self.runner = runner
+
+    def redhat_compat(self):
+        pass
+    
+    def debian_compat(self):
+        self.runner.run("apt-get -qq update -y")
+
+#----------------------------------------------------------------------------------------------
+
+class Setup(OnPlatform):
+    def __init__(self, nop=False):
+        OnPlatform.__init__(self)
+        self.runner = Runner(nop)
+        self.stages = [0]
+        self.platform = Platform()
+        self.os = self.platform.os
+        self.dist = self.platform.dist
+        self.ver = self.platform.os_ver
+        
+        if self.has_command("python"):
+            self.python = "python"
+        elif self.has_command("python2"):
+            self.python = "python2"
+        elif self.has_command("python3"):
+            self.python = "python3"
+
+        if self.os == 'macosx':
+            # this is required because osx pip installed are done with --user
+            os.environ["PATH"] = os.environ["PATH"] + ':' + '$HOME/Library/Python/2.7/bin'
```

Comment:
I'm not sure this would work on macOS. `conda` for instance prepends its paths to PATH, so this operation would appear after the `conda` PATHs.

Reply:
Well, this is actually not very relevant to our case as we decided to avoid python2.
The purpose of this shim was to allow you to execute stuff that was put by "pip install --user" into $HOME/Library/Python/2.7/bin (e.g., virtualenv, pipenv).